### ISSUE TITLE
Some rules not working with static method calls

### ISSUE THREAD

**Issue by movedoa** [Riverside-Software/sonar-openedge#1187]
3.1.0

The "class name casing" rule and the "missing parameter mode in method calls" rule dont work on static method calls.
Also the Missing parameter mode rule doesn't seem to work when a parameter is ?.

I have only tested this in SonarLint.

```
CLASS FooClass:

   METHOD PRIVATE VOID Foo1(
   ):
      DEFINE VARIABLE lcFoo AS CHARACTER NO-UNDO.
      DEFINE VARIABLE liFoo AS INTEGER NO-UNDO.

      // #### These work ###
      DEFINE VARIABLE loFooClass AS Fooclass NO-UNDO. // Case of variable declaration doesn't match case of compiled class FooClass

      THIS-OBJECT:Foo2("1":U, lcFoo). // INPUT-OUTPUT missing
      THIS-OBJECT:Foo2(1, liFoo). // INPUT-OUTPUT missing

      FooClass:StaticFoo2("1":U, lcFoo). // INPUT-OUTPUT missing
      FooClass:StaticFoo2(1, liFoo). // INPUT-OUTPUT missing

      THIS-OBJECT:Foo2("1":U, INPUT-OUTPUT lcFoo). // No issue, ok
      THIS-OBJECT:Foo2(1, INPUT-OUTPUT liFoo). // No issue, ok

      // #### These dont work ###
      THIS-OBJECT:Foo2(?, lcFoo). // No Issue shown, should be INPUT-OUTPUT missing
      THIS-OBJECT:Foo2(?, liFoo). // No Issue shown, should be INPUT-OUTPUT missing

      Fooclass:StaticFoo2("1":U, INPUT-OUTPUT lcFoo). // No Issue shown, should be Class name casing
      Fooclass:StaticFoo2(1, INPUT-OUTPUT  liFoo). // No Issue shown, should be Class name casing

      Fooclass:StaticFoo2("1":U, lcFoo). // No Issue shown, should be both Class name casing and INPUT-OUTPUT missing
      Fooclass:StaticFoo2(1, liFoo). // No Issue shown, should be both Class name casing and INPUT-OUTPUT missing
   END METHOD.

   METHOD PRIVATE VOID Foo2(
      pcFoo1 AS CHARACTER,
      INPUT-OUTPUT pcFoo2 AS CHARACTER
   ):
      IF pcFoo1 <> ? THEN DO:
         pcFoo2 = "Foo":U.
      END.
   END METHOD.

   METHOD PRIVATE VOID Foo2(
      piFoo1 AS INTEGER,
      INPUT-OUTPUT piFoo2 AS INTEGER
   ):
      IF piFoo1 <> ? THEN DO:
         piFoo2 = 1.
      END.
   END METHOD.

   METHOD PRIVATE STATIC VOID StaticFoo2(
      pcFoo1 AS CHARACTER,
      INPUT-OUTPUT pcFoo2 AS CHARACTER
   ):
      IF pcFoo1 <> ? THEN DO:
         pcFoo2 = "Foo":U.
      END.
   END METHOD.

   METHOD PRIVATE STATIC VOID StaticFoo2(
      piFoo1 AS INTEGER,
      INPUT-OUTPUT piFoo2 AS INTEGER
   ):
      IF piFoo1 <> ? THEN DO:
         piFoo2 = 1.
      END.
   END METHOD.

END CLASS.

```

**Comment by gquerret** [comment]
Fixed in develop, will be available in next release
